pkg/controller/storagecluster/reconcile.go

```diff
@@ -850,6 +852,18 @@ func determineFailureDomain(sc *ocsv1.StorageCluster) string {
 	return failureDomain
 }
 
+// validateMetadataPvc checks the MetadataPvc of the given
+// StorageCluster for completeness and correctness
+func validateMetadataPvc(sc *ocsv1.StorageCluster) error {
```

Comment:
Looking at this again, this should not be its own function. This should be part of the `validateStorageDeviceSet()` function, with an `if` to check if the metadataPVCTemplate is set. Similarly the unit tests should be updated to reflect that.

Reply:
The reason for not integrating this check into `validateStorageDeviceSet()`.
- When on using only DataPVCTemplate , the below statements in reconcile.go would return an error 

```
	if !instance.Spec.ExternalStorage.Enable {
		err = r.validateStorageDeviceSets(instance)
		if err != nil {
			reqLogger.Error(err, "Failed to validate StorageDeviceSets")
			return reconcile.Result{}, err
		}
	}
```

Reply:
Validation for metadatapvc template has been integrated with `validateStorageDeviceSet()` method